src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -542,5 +545,54 @@ public virtual Task<string> CopyNupkgAsync(string nupkgFilePath, CancellationTok
         {
             throw new NotImplementedException();
         }
+
+        public async Task<IReadOnlyList<Signature>> GetSignaturesAsync(CancellationToken token)
+        {
+            var signatures = new List<Signature>();
+
+            var files = await GetFilesAsync(token);
+            var sigFile = files.FirstOrDefault(e => StringComparer.Ordinal.Equals(e, "testsigned/signed.json"));
+
+            if (sigFile != null)
+            {
+                using (var stream = await GetStreamAsync(sigFile, token))
+                using (var reader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(reader))
+                {
+                    var json = JObject.Load(jsonReader);
+
+                    foreach (var sigEntry in json.Value<JArray>("signatures"))
+                    {
+                        Enum.TryParse<SignatureTrust>(sigEntry["trust"].Value<string>(), ignoreCase: true, result: out var testTrust);
+                        Enum.TryParse<SignatureType>(sigEntry["type"].Value<string>(), ignoreCase: true, result: out var sigType);
+
+                        signatures.Add(new Signature()
+                        {
+                            DisplayName = sigEntry.Value<string>("name"),
+                            TestTrust = testTrust,
+                            Type = sigType,
+                        });
+                    }
+                }
+            }
+
+            return signatures;
+        }
+
+        public Task<SignManifest> GetSignManifestAsync(CancellationToken token)
+        {
+            return CreateManifestAsync(token);
+        }
+
+        public Task<SignManifest> CreateManifestAsync(CancellationToken token)
+        {
+            return Task.FromResult(SignManifest.Create(new Dictionary<string, string>(), Enumerable.Empty<SignManifestFileEntry>()));
+        }
+
+        public async Task<bool> IsSignedAsync(CancellationToken token)
+        {
+            var signatures = await GetSignaturesAsync(token);
```

Comment:
should we cache these signatures instead of reading them from files every time?

Reply:
@jainaashish sounds like a good idea for real signing, the current implementation is only for test signing so I've avoided any perf optimizations. I'm looking for feedback on the API surface area and interfaces here.